### PR TITLE
Migrated some Legacy Tests to Integration/Unit Tests

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Migrated some Legacy Tests to Integration/Unit Tests
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to #23462
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25127)

### CODE REVIEW

File: tests/Integration/Classes/controller/AdminControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+            ['AdminCartsController'],
+            ['AdminImagesController'],
+            ['AdminShopUrlController'],
+            ['AdminStatesController'],
+            ['AdminStatsController'],
+            ['AdminLegacyLayoutController'],
+        ];
+    }
+
+    protected function declareRequiredConstants(): void
```
Comment (sowbiba) [inline tests/Integration/Classes/controller/AdminControllerTest.php]: why not private ?

File: tests/Integration/Classes/controller/AdminControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        }
+        if (!defined('_PS_VERSION_')) {
+            define('_PS_VERSION_', '1.7');
+        }
+        if (!defined('_PS_ADMIN_DIR_')) {
+            define('_PS_ADMIN_DIR_', '');
+        }
+    }
+
+    protected function getMockTranslator(): Translator
```
Comment (sowbiba) [inline tests/Integration/Classes/controller/AdminControllerTest.php]: why not private ?

File: tests/Integration/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php [hunk 3]
```diff
@@ -41,32 +43,35 @@ class YamlRoutesInModuleTest extends KernelTestCase
...
     }
 
     /**
      * {@inheritdoc}
      */
-    protected function tearDown()
+    protected function tearDown(): void
     {
-        HelperModule::removeModule('demo');
+        $dirResources = dirname(__DIR__, 4);
```
Comment (sowbiba) [inline tests/Integration/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php:65]: don't we need to call parent::tearDown to shutdown kernel ?

File: tests/Resources/modules/demo/vendor/composer/autoload_static.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        'P' => 
+        array (
+            'PsTest\\' => 7,
+        ),
+    );
+
+    public static $prefixDirsPsr4 = array (
+        'PsTest\\' => 
+        array (
+            0 => __DIR__ . '/../..' . '/src',
```
Comment (sowbiba) [inline tests/Resources/modules/demo/vendor/composer/autoload_static.php:19]: why not using dirname here ?

File: tests/Integration/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php (hunk 3 above)
Comment (Progi1984) [reply tests/Integration/PrestaShopBundle/Routing/Converter/RoutingCacheKeyGeneratorTest.php:65]: Nope we add & remove module at the fly.

File: tests/Resources/modules/demo/vendor/composer/autoload_static.php (hunk 4 above)
Comment (Progi1984) [reply tests/Resources/modules/demo/vendor/composer/autoload_static.php:19]: it's composer generated.

**Review by sowbiba — DISMISSED** [review]
LGTM
Thanks @Progi1984

**Review by matthieu-rolland — DISMISSED** [review]

File: src/Core/Localization/CLDR/Currency.php [hunk 5]
```diff
@@ -140,9 +140,15 @@ public function getDecimalDigits()
      */
     public function getDisplayName($countContext = CurrencyInterface::DISPLAY_NAME_COUNT_DEFAULT)
     {
+        dump($this->displayNames);
```
Comment (sowbiba) [inline src/Core/Localization/CLDR/Currency.php]: forgotten ?

File: tests/Integration/PrestaShopBundle/Model/Product/AdminModelAdapterTest.php [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        $product->name = 'Product name';
+
+        return $product;
+    }
+
+    protected function setUp(): void
+    {
+        self::bootKernel();
+
+        \Context::getContext()->shop = new \Shop(1);
```
Comment (sowbiba) [inline tests/Integration/PrestaShopBundle/Model/Product/AdminModelAdapterTest.php]: Please add `use` statements

**Review by sowbiba — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
I'm a bit worried about the modification in `Currency` class

It highlights that we don't fully control/master/understand the expected behaviour of this part of the code as discussed with @PierreRambaud it would deserve to add lots of unit test and to dig into the exact expected technical behaviour, but that's not the point of this PR

**Pull request merged**